src/rich_click/rich_command.py

```diff
@@ -13,21 +16,61 @@ class RichCommand(click.Command):
     """
 
     standalone_mode = False
+    context_class: ClassVar[Type[RichContext]] = RichContext
+
+    def __init__(self, *args, **kwargs):
+        """Create Rich Command.
+
+        Accepts same arguments as Click Command.
+
+        Docs reference: https://click.palletsprojects.com/
+        """
+        super().__init__(*args, **kwargs)
+        self._formatter: Optional[RichHelpFormatter] = None
+
+    @property
+    def console(self):
+        """Rich Console.
+
+        This is a separate instance from the help formatter that allows full control of the
+        console configuration.
+
+        See `rich_config` decorator for how to apply the settings.
+        """
+        return self.context_settings.get("rich_console")
+
+    @property
+    def help_config(self):
+        """Rich Help Configuration."""
+        return self.context_settings.get("rich_help_config")
+
+    @property
+    def formatter(self):
+        """Rich Help Formatter.
+
+        This is separate instance from the formatter used to display help,
+        but is created from the same `RichHelpConfiguration`. Currently only used
+        for error reporting.
+        """
+        return self._formatter
 
     def main(self, *args, standalone_mode: bool = True, **kwargs):
+        formatter = self._formatter = RichHelpFormatter(config=self.help_config)
```

Comment:
Not sure the formatter is worth exposing here. It's only needed in the exception handling at this point as the formatter used for help generation is created by click internally. I only exposed it as a possible convenience for users. Maybe the Rich console is enough.